### ISSUE TITLE
Support K2 / KSP2

### ISSUE THREAD

**Issue by mcarleio** [Dyescape/kotlin-maven-symbol-processing#107]
Hi there!

I know there is still a lot of stuff ongoing and KSP2 still in beta - just curious if you are planning to support the new version or not :)

Best

Marcel

**Comment by amitojduggal** [comment]
@mcarleio I added an MR with an upgrade to Kotlin 2.x and Ksp 2.0. Looking forward to be able to use Konvert with 2.x Kotlin :)

https://github.com/Dyescape/kotlin-maven-symbol-processing/pull/109 

@MrDienns @evanchooly possible to have a look at open MR https://github.com/Dyescape/kotlin-maven-symbol-processing/pull/109 
 ?

**Comment by amitojduggal** [comment]
@mcarleio how about creating a fork for this, seems the work on this project has been stopped?

**Comment by mcarleio** [comment]
I do not have the time nor energy to maintain a fork. If you have that, it would help to keep maven support in Konvert. Otherwise, worst case, I need to abandon maven support in Konvert with the next release.

**Comment by MrDienns** [comment]
Hi there! Apologies for the late reply. We suggest forking the project or copying over the source files as we've migrated everything internally to Gradle and stopped using Maven. We will archive the project soon.

**Comment by amitojduggal** [comment]
Thanks for letting us know @MrDienns .

@mcarleio I'm happy to do so and this way we can keep Konvert's support for maven.
Once we move to Kotlin 2.x, we can continue with minimal changes for a longer period of time.

I am checking if we can create a fork under my organisations's github, else I will create a fork on my own account and let you know. 

Looking forward to working on this.

**Comment by evanchooly** [comment]
I wish I could help with a release but I'm just a rando here, too.

**Comment by mkoertgen** [comment]
I would offer help in forking or adopting the project.

**Comment by amitojduggal** [comment]
@mkoertgen big thanks for offering help, lets do it.
How shall we proceed? shall we create this under a new artifact/group id, or keep the same, any suggestions?

**Comment by mkoertgen** [comment]
Yes, exactly. I would suggest to just switch the groupId and publish it to
- https://central.sonatype.com/
(just like current deployment)

Using central instead of other repositories (e.g. [GitHub Maven Packages](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry)) makes it easier to consume (no additional repositories in the pom necessary).

This would be the docs for central: https://central.sonatype.org/publish/publish-portal-maven/#publishing

**Comment by mcarleio** [comment]
@amitojduggal, @mkoertgen : Did you manage to create a fork? If so, is it working with KSP2? Then I would switch to that version in Konvert (i.e. maven example project and maven usage documentation).

**Comment by mkoertgen** [comment]
Sorry to pull back, but my needs went away since we used the kapt-kotlin-workaround months ago.

**Comment by mcarleio** [comment]
Whoever is interested in using KSP2 with maven:

Based on the documentation and with a bit trial and error, I managed to create a maven plugin [ksp-maven-plugin](https://github.com/mcarleio/ksp-maven-plugin) which you can include in your maven projects like this:

```xml
<plugin>
    <groupId>io.mcarle</groupId>
    <artifactId>ksp-maven-plugin</artifactId>
    <version>2.3.0-1</version>
    <executions>
        <execution>
            <goals>
                <goal>ksp</goal>
            </goals>
        </execution>
    </executions>
    <configuration>
        <!-- Optional: Define processor options -->
        <processorOptions>
            <option>key=value</option>
        </processorOptions>
    </configuration>
    <dependencies>
        <!-- Add your KSP processors here, e.g. konvert -->
        <dependency>
            <groupId>io.mcarle</groupId>
            <artifactId>konvert</artifactId>
            <version>4.4.0</version>
        </dependency>
    </dependencies>
</plugin>
```

Have a look into the [example project](https://github.com/mcarleio/konvert/blob/main/example/pom.xml#L75) of [konvert](https://github.com/mcarleio/konvert) to see it working in action.
